server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -152,10 +164,18 @@ public boolean isBlockEligible(String blockId) {
 
   @Override
   public ImmutableList<Block> getAllActiveBlocks() {
-    if (allBlockList == null) {
-      allBlockList = getBlocks(this::showBlock);
+    if (allActiveBlockList == null) {
+      allActiveBlockList = getBlocks(this::showBlock, /* includeHidden= */ false);
     }
-    return allBlockList;
+    return allActiveBlockList;
+  }
+
+  @Override
+  public ImmutableList<Block> getAllHiddenBlocks() {
+    if (allHiddenBlockList == null) {
+      allHiddenBlockList = getBlocks(this::showBlock, /* includeHidden= */ true);
```

Comment:
*question*: Does `includeHidden` mean that it returns active _and_ hidden blocks, or only hidden blocks?

If it's active and hidden, then calls above this might get active blocks unexpectedly since `getAllHiddenBlocks()` implies you're only getting the hidden ones.

*suggestion*:
If it's only hidden blocks, then we should rename the parameter to `onlyHidden` so it's more clear.

Reply:
See also the note below about combining `includeHidden` with the `includeIfTrue` predicate.